src/chains.rs

```diff
@@ -766,6 +799,7 @@ impl<'a> ChainFormatterShared<'a> {
             Cow::from("")
         } else {
             // Use new lines.
+            // TODO question new variant or max width err
             if context.force_one_line_chain.get() {
                 return None;
             }
```

Comment:
Unlike `write_list` where we have list of `RewriteResult` as item, `join_rewrites` concats list of `String`. 
(see `rewrites: Vec<String>,` in `struct ChainFormatterShared`)  

So the only point that returns `None` is when we cannot make this chain fit in one line. I think we can still represent this error as max width error but passing span of children (that we manually extracted from caller) is necessary.

Reply:
Thanks for the additional explanation. If that's the case, then I'd be fine with leaving `join_rewrite` as returning `Option<String>`. My previous comment was made with the assumption that this worked similar to `write_list`.